soh/src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -223,13 +285,13 @@ void func_809DF7D8(EnCow* this, GlobalContext* globalCtx) {
         this->actor.flags &= ~ACTOR_FLAG_16;
         Message_CloseTextbox(globalCtx);
         this->actionFunc = func_809DF778;
-        func_8002F434(&this->actor, globalCtx, GI_MILK, 10000.0f, 100.0f);
+        func_8002F434(&this->actor, globalCtx, EnCow_GetItemFromCow(this, globalCtx, true), 10000.0f, 100.0f);
     }
 }
 
 void func_809DF870(EnCow* this, GlobalContext* globalCtx) {
     if ((Message_GetState(&globalCtx->msgCtx) == TEXT_STATE_EVENT) && Message_ShouldAdvance(globalCtx)) {
-        if (Inventory_HasEmptyBottle()) {
+        if (EnCow_GetItemFromCow(this, globalCtx, false) != ITEM_NONE) {
```

Comment:
Similar to `func_8002F434()` above, I'd suggest keeping the vanilla code in and only run rando specific code when we're rando'd. Afterwards, you can rename the function to specify it's a rando function as well.